src/battle_anim_new.c

```diff
@@ -7895,3 +7896,28 @@ void AnimTask_TerrainPulse(u8 taskId)
     }
     DestroyAnimVisualTask(taskId);
 }
+
+void AnimTask_AffectionHangedOn(u8 taskId)
+{
+    int side = GetBattlerSide(gBattleAnimTarget);
+    struct Pokemon *party = (side == B_SIDE_PLAYER) ? gPlayerParty : gEnemyParty;
+
+    switch (GetMonFriendshipScore(&party[gBattlerPartyIndexes[gBattleAnimTarget]]))
```

Comment:
Isn't this function doing sth like
case x:
gBattleAnimArgs[0] = x;

Why not just
gBattleAnimArgs[0] = GetMonFriendshipScore(&party[gBattlerPartyIndexes[gBattleAnimTarget]]); ?

Reply:
I just needed to shut down the function if the value returned by `GetMonFriendshipScore` was smaller than `FRIENDSHIP_100_TO_149`, buuut I think you're right, that's probably not a concern because that can never actually happen. Will fix.